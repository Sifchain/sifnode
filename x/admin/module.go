package admin

import (
	"encoding/json"
	"fmt"

	"github.com/Sifchain/sifnode/x/admin/client/cli"

	sdkclient "github.com/cosmos/cosmos-sdk/client"
	"github.com/cosmos/cosmos-sdk/codec"
	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/cosmos/cosmos-sdk/types/module"
	"github.com/gorilla/mux"
	"github.com/grpc-ecosystem/grpc-gateway/runtime"
	"github.com/spf13/cobra"
	abci "github.com/tendermint/tendermint/abci/types"

	//"github.com/Sifchain/sifnode/x/tokenregistry/handler"
	"github.com/Sifchain/sifnode/x/admin/keeper"
	"github.com/Sifchain/sifnode/x/admin/types"
)

var (
	ModuleName                       = types.ModuleName
	_          module.AppModule      = AppModule{}
	_          module.AppModuleBasic = AppModuleBasic{}
)

// AppModuleBasic defines the basic application module.
type AppModuleBasic struct{}

func (b AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
	types.RegisterInterfaces(registry)
}

var _ module.AppModuleBasic = AppModuleBasic{}

// Name returns the module's name.
func (AppModuleBasic) Name() string {
	return ModuleName
}

// RegisterCodec registers the module's types for the given codec.
func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) { //nolint
	types.RegisterLegacyAminoCodec(cdc)
}

// DefaultGenesis returns default genesis state as raw bytes.
func (b AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
	return cdc.MustMarshalJSON(&types.GenesisState{})
}

// ValidateGenesis performs genesis state validation.
func (b AppModuleBasic) ValidateGenesis(marshaler codec.JSONCodec, _ sdkclient.TxEncodingConfig, message json.RawMessage) error {
	var data types.GenesisState
	if message != nil {
		err := marshaler.UnmarshalJSON(message, &data)
		if err != nil {
			return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
		}
	}
	return nil
}

func (b AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx sdkclient.Context, mux *runtime.ServeMux) {
	//err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
	//if err != nil {
	//	panic(err)
	//}
}

// RegisterRESTRoutes registers the REST routes.
func (b AppModuleBasic) RegisterRESTRoutes(ctx sdkclient.Context, router *mux.Router) {
	// rest.RegisterRESTRoutes(ctx, router)
}

// GetTxCmd returns the root tx command.
func (b AppModuleBasic) GetTxCmd() *cobra.Command {
	return cli.GetTxCmd()
}

// GetQueryCmd returns no root query command.
func (AppModuleBasic) GetQueryCmd() *cobra.Command {
	return cli.GetQueryCmd()
}

//____________________________________________________________________________

// AppModuleSimulation defines the module simulation functions.
type AppModuleSimulation struct{}

// AppModule implements an application module.
type AppModule struct {
	AppModuleBasic
	AppModuleSimulation
	Keeper keeper.Keeper
	Codec  *codec.Codec
}

func (am AppModule) RegisterServices(cfg module.Configurator) {
	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.Keeper))
	types.RegisterQueryServer(cfg.QueryServer(), keeper.NewQueryServer(am.Keeper))
	//m := keeper.NewMigrator(am.Keeper)
	//err := cfg.RegisterMigration(types.ModuleName, 0, m.InitialMigration)
	//if err != nil {
	//	panic(err)
	//}
}

// NewAppModule creates a new AppModule object
func NewAppModule(keeper keeper.Keeper, cdc *codec.Codec) AppModule {
	return AppModule{
		AppModuleBasic:      AppModuleBasic{},
		AppModuleSimulation: AppModuleSimulation{},
		Keeper:              keeper,
		Codec:               cdc,
	}
}

// Name returns the module's name.
func (AppModule) Name() string {
	return ModuleName
}

// RegisterInvariants registers the module invariants.
func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {
}

// Route returns the message routing key for the module.
func (am AppModule) Route() sdk.Route {
	return sdk.NewRoute(types.RouterKey, am.NewHandler())
}

// NewHandler returns an sdk.Handler for the module.
func (am AppModule) NewHandler() sdk.Handler {
	return keeper.NewHandler(am.Keeper)
}

// QuerierRoute returns the module's querier route name.
func (AppModule) QuerierRoute() string {
	return types.QuerierRoute
}

// Deprecated: LegacyQuerierHandler use RegisterServices
func (am AppModule) LegacyQuerierHandler(_ *codec.LegacyAmino) sdk.Querier { //nolint
	return keeper.NewLegacyQuerier(am.Keeper)
}

// InitGenesis performs genesis initialization. It returns
// no validator updates.
func (am AppModule) InitGenesis(ctx sdk.Context, marshaler codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
	var genesisState types.GenesisState
	marshaler.MustUnmarshalJSON(data, &genesisState)
	return am.Keeper.InitGenesis(ctx, genesisState)
}

// ExportGenesis returns the exported genesis state as raw bytes.
func (am AppModule) ExportGenesis(ctx sdk.Context, marshaler codec.JSONCodec) json.RawMessage {
	return marshaler.MustMarshalJSON(am.Keeper.ExportGenesis(ctx))
}

// BeginBlock returns the begin blocker.
func (AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}

// EndBlock returns the end blocker. It returns no validator updates.
func (am AppModule) EndBlock(_ sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
	return nil
}

func (AppModule) ConsensusVersion() uint64 { return 3 }
