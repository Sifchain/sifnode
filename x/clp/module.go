package clp

import (
	"encoding/json"
	"github.com/Sifchain/sifnode/x/clp/client/cli"
	"github.com/Sifchain/sifnode/x/clp/client/rest"
	"github.com/Sifchain/sifnode/x/clp/types"
	"github.com/gorilla/mux"
	"github.com/spf13/cobra"

	abci "github.com/tendermint/tendermint/abci/types"

	"github.com/Sifchain/sifnode/x/clp/keeper"
	"github.com/cosmos/cosmos-sdk/client/context"
	"github.com/cosmos/cosmos-sdk/codec"
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/cosmos/cosmos-sdk/types/module"
)

// Type check to ensure the interface is properly implemented
var (
	_ module.AppModule      = AppModule{}
	_ module.AppModuleBasic = AppModuleBasic{}
)

// AppModuleBasic defines the basic application module used by the clp module.
type AppModuleBasic struct{}

// Name returns the clp module's name.
func (AppModuleBasic) Name() string {
	return ModuleName
}

// RegisterCodec registers the clp module's types for the given codec.
func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
	RegisterCodec(cdc)
}

// DefaultGenesis returns default genesis state as raw bytes for the clp
// module.
func (AppModuleBasic) DefaultGenesis() json.RawMessage {
	return ModuleCdc.MustMarshalJSON(DefaultGenesisState())
}

// ValidateGenesis performs genesis state validation for the clp module.
func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
	//var data GenesisState
	//err := ModuleCdc.UnmarshalJSON(bz, &data)
	//if err != nil {
	//	return fmt.Errorf("failed to unmarshal %s genesis state: %w", ModuleName, err)
	//}
	return ValidateGenesis(GenesisState{})
}

// RegisterRESTRoutes registers the REST routes for the clp module.
func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
	rest.RegisterRoutes(ctx, rtr)
}

// GetTxCmd returns the root tx command for the clp module.
func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
	return cli.GetTxCmd(cdc)
}

// GetQueryCmd returns no root query command for the clp module.
func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
	return cli.GetQueryCmd(StoreKey, cdc)
}

//____________________________________________________________________________

// AppModule implements an application module for the clp module.
type AppModule struct {
	AppModuleBasic

	keeper     keeper.Keeper
	bankKeeper types.BankKeeper
}

// NewAppModule creates a new AppModule object
func NewAppModule(k keeper.Keeper, bankKeeper types.BankKeeper) AppModule {
	return AppModule{
		AppModuleBasic: AppModuleBasic{},
		keeper:         k,
		bankKeeper:     bankKeeper,
	}
}

// Name returns the clp module's name.
func (AppModule) Name() string {
	return ModuleName
}

// RegisterInvariants registers the clp module invariants.
func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}

// Route returns the message routing key for the clp module.
func (AppModule) Route() string {
	return RouterKey
}

// NewHandler returns an sdk.Handler for the clp module.
func (am AppModule) NewHandler() sdk.Handler {
	return NewHandler(am.keeper)
}

// QuerierRoute returns the clp module's querier route name.
func (AppModule) QuerierRoute() string {
	return QuerierRoute
}

// NewQuerierHandler returns the clp module sdk.Querier.
func (am AppModule) NewQuerierHandler() sdk.Querier {
	return NewQuerier(am.keeper)
}

// InitGenesis performs genesis initialization for the clp module. It returns
// no validator updates.
func (am AppModule) InitGenesis(ctx sdk.Context, data json.RawMessage) []abci.ValidatorUpdate {
	var genesisState GenesisState
	ModuleCdc.MustUnmarshalJSON(data, &genesisState)
	return InitGenesis(ctx, am.keeper, am.bankKeeper, genesisState)
}

// ExportGenesis returns the exported genesis state as raw bytes for the clp
// module.
func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
	gs := ExportGenesis(ctx, am.keeper)
	return ModuleCdc.MustMarshalJSON(gs)
}

// BeginBlock returns the begin blocker for the clp module.
func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
	BeginBlocker(ctx, req, am.keeper)
}

// EndBlock returns the end blocker for the clp module. It returns no validator
// updates.
func (AppModule) EndBlock(_ sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
	return []abci.ValidatorUpdate{}
}
