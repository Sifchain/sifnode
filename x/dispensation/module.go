package dispensation

import (
	"encoding/json"
	"fmt"

	"github.com/Sifchain/sifnode/x/dispensation/client/rest"

	"github.com/cosmos/cosmos-sdk/client"
	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
	"github.com/cosmos/cosmos-sdk/types/module"
	"github.com/gorilla/mux"
	"github.com/grpc-ecosystem/grpc-gateway/runtime"
	"github.com/spf13/cobra"

	"github.com/Sifchain/sifnode/x/dispensation/client/cli"

	abci "github.com/tendermint/tendermint/abci/types"

	"github.com/cosmos/cosmos-sdk/codec"
	sdk "github.com/cosmos/cosmos-sdk/types"

	"github.com/Sifchain/sifnode/x/dispensation/keeper"
	"github.com/Sifchain/sifnode/x/dispensation/types"
)

// Type check to ensure the interface is properly implemented
var (
	_ module.AppModule      = AppModule{}
	_ module.AppModuleBasic = AppModuleBasic{}
)

// AppModuleBasic defines the basic application module used by the dispensation module.
type AppModuleBasic struct{}

func (AppModuleBasic) Name() string {
	return types.ModuleName
}

func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) { //nolint
	types.RegisterLegacyAminoCodec(cdc)
}

// RegisterInterfaces registers the module's interface types
func (b AppModuleBasic) RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
	types.RegisterInterfaces(registry)
}

// DefaultGenesis returns default genesis state as raw bytes for the dispensation module.
func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
	return cdc.MustMarshalJSON(types.DefaultGenesisState())
}

// RegisterCodec registers the dispensation module's types for the given codec.
func (AppModuleBasic) RegisterCodec(cdc *codec.LegacyAmino) { //nolint
	RegisterCodec(cdc)
}

// ValidateGenesis performs genesis state validation for the dispensation module.
func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
	var data GenesisState
	err := cdc.UnmarshalJSON(bz, &data)
	if err != nil {
		return fmt.Errorf("failed to unmarshal %s genesis state: %w", ModuleName, err)
	}
	return ValidateGenesis(data)
}

// RegisterRESTRoutes registers the REST routes for the dispensation module.
func (AppModuleBasic) RegisterRESTRoutes(ctx client.Context, rtr *mux.Router) {
	rest.RegisterRoutes(ctx, rtr)
}

func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
	//types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
}

// GetTxCmd returns the root tx command for the dispensation module.
func (AppModuleBasic) GetTxCmd() *cobra.Command {
	return cli.GetTxCmd()
}

// GetQueryCmd returns no root query command for the dispensation module.
func (AppModuleBasic) GetQueryCmd() *cobra.Command {
	return cli.GetQueryCmd(StoreKey)
}

//____________________________________________________________________________

// AppModule implements an application module for the dispensation module.
type AppModule struct {
	AppModuleBasic

	keeper        keeper.Keeper
	bankKeeper    types.BankKeeper
	accountKeeper types.AccountKeeper
}

func (am AppModule) LegacyQuerierHandler(amino *codec.LegacyAmino) sdk.Querier { //nolint
	return keeper.NewLegacyQuerier(am.keeper)
}

func (am AppModule) RegisterServices(cfg module.Configurator) {
	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.keeper))
	querier := keeper.NewQuerier(am.keeper)
	types.RegisterQueryServer(cfg.QueryServer(), querier)
	m := keeper.NewMigrator(am.keeper)
	err := cfg.RegisterMigration(types.ModuleName, 1, m.MigrateToVer2)
	if err != nil {
		panic(err)
	}
}

// NewAppModule creates a new AppModule object
func NewAppModule(k keeper.Keeper, bankKeeper types.BankKeeper, accountKeeper types.AccountKeeper) AppModule {
	return AppModule{
		AppModuleBasic: AppModuleBasic{},
		keeper:         k,
		bankKeeper:     bankKeeper,
		accountKeeper:  accountKeeper,
	}
}

// Name returns the dispensation module's name.
func (AppModule) Name() string {
	return ModuleName
}

// RegisterInvariants registers the dispensation module invariants.
func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}

// Route returns the message routing key for the dispensation module.
func (am AppModule) Route() sdk.Route {
	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper))
}

// NewHandler returns an sdk.Handler for the dispensation module.
func (am AppModule) NewHandler() sdk.Handler {
	return NewHandler(am.keeper)
}

// QuerierRoute returns the dispensation module's querier route name.
func (AppModule) QuerierRoute() string {
	return QuerierRoute
}

// NewQuerierHandler returns the dispensation module sdk.Querier.
func (am AppModule) NewQuerierHandler() sdk.Querier {
	return keeper.NewLegacyQuerier(am.keeper)
}

// InitGenesis performs genesis initialization for the dispensation module. It returns
// no validator updates.
func (am AppModule) InitGenesis(ctx sdk.Context, codec codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
	var genesisState GenesisState
	codec.MustUnmarshalJSON(data, &genesisState)
	return InitGenesis(ctx, am.keeper, genesisState)
}

// ExportGenesis returns the exported genesis state as raw bytes for the dispensation
// module.
func (am AppModule) ExportGenesis(ctx sdk.Context, codec codec.JSONCodec) json.RawMessage {
	gs := ExportGenesis(ctx, am.keeper)
	return codec.MustMarshalJSON(&gs)
}

// BeginBlock returns the begin blocker for the dispensation module.
func (am AppModule) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
	BeginBlocker(ctx, am.keeper)
}

// EndBlock returns the end blocker for the dispensation module. It returns no validator
// updates.
func (AppModule) EndBlock(_ sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
	return []abci.ValidatorUpdate{}
}

func (AppModule) ConsensusVersion() uint64 { return 2 }
