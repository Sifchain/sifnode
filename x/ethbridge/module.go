package ethbridge

import (
	"encoding/json"

	"github.com/gorilla/mux"
	"github.com/spf13/cobra"

	abci "github.com/tendermint/tendermint/abci/types"

	"github.com/cosmos/cosmos-sdk/client/context"
	"github.com/cosmos/cosmos-sdk/codec"
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/cosmos/cosmos-sdk/types/module"
	"github.com/cosmos/cosmos-sdk/x/supply"

	"github.com/Sifchain/sifnode/x/ethbridge/client"
	"github.com/Sifchain/sifnode/x/ethbridge/types"
)

var (
	_ module.AppModule      = AppModule{}
	_ module.AppModuleBasic = AppModuleBasic{}
)

// AppModuleBasic defines the basic application module used by the ethbridge module.
type AppModuleBasic struct{}

var _ module.AppModuleBasic = AppModuleBasic{}

// Name returns the ethbridge module's name.
func (AppModuleBasic) Name() string {
	return ModuleName
}

// RegisterCodec registers the ethbridge module's types for the given codec.
func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
	types.RegisterCodec(cdc)
}

// DefaultGenesis returns default genesis state as raw bytes for the ethbridge
// module.
func (AppModuleBasic) DefaultGenesis() json.RawMessage {
	return nil
}

// ValidateGenesis performs genesis state validation for the ethbridge module.
func (AppModuleBasic) ValidateGenesis(_ json.RawMessage) error {
	return nil
}

// RegisterRESTRoutes registers the REST routes for the ethbridge module.
func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
	client.RegisterRESTRoutes(ctx, rtr, StoreKey)
}

// GetTxCmd returns the root tx command for the ethbridge module.
func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
	return client.GetTxCmd(StoreKey, cdc)
}

// GetQueryCmd returns no root query command for the ethbridge module.
func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
	return client.GetQueryCmd(StoreKey, cdc)
}

//____________________________________________________________________________

// AppModuleSimulation defines the module simulation functions used by the ethbridge module.
type AppModuleSimulation struct{}

// AppModule implements an application module for the ethbridge module.
type AppModule struct {
	AppModuleBasic
	AppModuleSimulation

	OracleKeeper  types.OracleKeeper
	SupplyKeeper  types.SupplyKeeper
	AccountKeeper types.AccountKeeper
	BridgeKeeper  Keeper
	Codec         *codec.Codec
}

// NewAppModule creates a new AppModule object
func NewAppModule(
	oracleKeeper types.OracleKeeper, supplyKeeper types.SupplyKeeper,
	accountKeeper types.AccountKeeper, bridgeKeeper Keeper,
	cdc *codec.Codec) AppModule {

	return AppModule{
		AppModuleBasic:      AppModuleBasic{},
		AppModuleSimulation: AppModuleSimulation{},

		OracleKeeper:  oracleKeeper,
		SupplyKeeper:  supplyKeeper,
		AccountKeeper: accountKeeper,
		BridgeKeeper:  bridgeKeeper,
		Codec:         cdc,
	}
}

// Name returns the ethbridge module's name.
func (AppModule) Name() string {
	return ModuleName
}

// RegisterInvariants registers the ethbridge module invariants.
func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
}

// Route returns the message routing key for the ethbridge module.
func (AppModule) Route() string {
	return RouterKey
}

// NewHandler returns an sdk.Handler for the ethbridge module.
func (am AppModule) NewHandler() sdk.Handler {
	return NewHandler(am.AccountKeeper, am.BridgeKeeper, am.Codec)
}

// QuerierRoute returns the ethbridge module's querier route name.
func (AppModule) QuerierRoute() string {
	return QuerierRoute
}

// NewQuerierHandler returns the ethbridge module sdk.Querier.
func (am AppModule) NewQuerierHandler() sdk.Querier {
	return NewQuerier(am.OracleKeeper, am.Codec)
}

// InitGenesis performs genesis initialization for the ethbridge module. It returns
// no validator updates.
func (am AppModule) InitGenesis(ctx sdk.Context, _ json.RawMessage) []abci.ValidatorUpdate {
	bridgeAccount := supply.NewEmptyModuleAccount(ModuleName, supply.Burner, supply.Minter)
	am.SupplyKeeper.SetModuleAccount(ctx, bridgeAccount)
	return nil
}

// ExportGenesis returns the exported genesis state as raw bytes for the ethbridge
// module.
func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
	return nil
}

// BeginBlock returns the begin blocker for the ethbridge module.
func (AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}

// EndBlock returns the end blocker for the ethbridge module. It returns no validator
// updates.
func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
	return nil
}
