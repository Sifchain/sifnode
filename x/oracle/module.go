package oracle

import (
	"encoding/json"

	"github.com/gorilla/mux"
	"github.com/spf13/cobra"

	abci "github.com/tendermint/tendermint/abci/types"

	"github.com/cosmos/cosmos-sdk/client/context"
	"github.com/cosmos/cosmos-sdk/codec"
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/cosmos/cosmos-sdk/types/module"
)

var (
	_ module.AppModule      = AppModule{}
	_ module.AppModuleBasic = AppModuleBasic{}
)

// AppModuleBasic defines the basic application module used by the oracle module.
type AppModuleBasic struct{}

var _ module.AppModuleBasic = AppModuleBasic{}

// Name returns the oracle module's name.
func (AppModuleBasic) Name() string {
	return ModuleName
}

// RegisterCodec registers the oracle module's types for the given codec.
func (AppModuleBasic) RegisterCodec(_ *codec.Codec) {}

// DefaultGenesis returns default genesis state as raw bytes for the oracle
// module.
func (AppModuleBasic) DefaultGenesis() json.RawMessage {
	return nil
}

// ValidateGenesis performs genesis state validation for the oracle module.
func (AppModuleBasic) ValidateGenesis(_ json.RawMessage) error {
	return nil
}

// RegisterRESTRoutes registers the REST routes for the oracle module.
func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
}

// GetTxCmd returns the root tx command for the oracle module.
func (AppModuleBasic) GetTxCmd(_ *codec.Codec) *cobra.Command {
	return nil
}

// GetQueryCmd returns no root query command for the oracle module.
func (AppModuleBasic) GetQueryCmd(_ *codec.Codec) *cobra.Command {
	return nil
}

//____________________________________________________________________________

// AppModuleSimulation defines the module simulation functions used by the oracle module.
type AppModuleSimulation struct{}

// AppModule implements an application module for the oracle module.
type AppModule struct {
	AppModuleBasic
	AppModuleSimulation

	keeper Keeper
}

// NewAppModule creates a new AppModule object
func NewAppModule(keeper Keeper) AppModule {
	return AppModule{
		AppModuleBasic:      AppModuleBasic{},
		AppModuleSimulation: AppModuleSimulation{},
		keeper:              keeper,
	}
}

// Name returns the oracle module's name.
func (AppModule) Name() string {
	return ModuleName
}

// RegisterInvariants registers the oracle module invariants.
func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
}

// Route returns the message routing key for the oracle module.
func (AppModule) Route() string {
	return RouterKey
}

// NewHandler returns an sdk.Handler for the oracle module.
func (am AppModule) NewHandler() sdk.Handler {
	return nil
}

// QuerierRoute returns the oracle module's querier route name.
func (AppModule) QuerierRoute() string {
	return ""
}

// NewQuerierHandler returns the oracle module sdk.Querier.
func (am AppModule) NewQuerierHandler() sdk.Querier {
	return nil
}

// InitGenesis performs genesis initialization for the oracle module. It returns
// no validator updates.
func (am AppModule) InitGenesis(_ sdk.Context, _ json.RawMessage) []abci.ValidatorUpdate {
	return nil
}

// ExportGenesis returns the exported genesis state as raw bytes for the oracle
// module.
func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
	return nil
}

// BeginBlock returns the begin blocker for the oracle module.
func (AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}

// EndBlock returns the end blocker for the oracle module. It returns no validator
// updates.
func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
	return nil
}
