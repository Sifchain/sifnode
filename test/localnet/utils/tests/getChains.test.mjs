import { getChains } from "../getChains.mjs";

import { createRequire } from "module";
const require = createRequire(import.meta.url);
const defaultChains = require("../../config/chains.json");

test("gets chains", () => {
  const chains = getChains({ chains: defaultChains });

  expect(chains).toMatchSnapshot();
});
