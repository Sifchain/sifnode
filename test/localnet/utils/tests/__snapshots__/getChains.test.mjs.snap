// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets chains 1`] = `
Object {
  "akash": Object {
    "binary": "akash",
    "binaryRelativePath": "akash_0.14.0_linux_amd64/akash",
    "binaryUrl": "https://github.com/ovrclk/akash/releases/download/v0.14.0/akash_0.14.0_linux_amd64.zip",
    "chainId": "akash-testnet-6",
    "denom": "uakt",
    "devnet-1": Object {
      "channelId": 115,
      "counterpartyChannelId": 65,
    },
    "disabled": true,
    "fees": 5000,
    "home": "/tmp/localnet/config/akash/akash-testnet-6",
    "node": "http://147.75.32.35:26657",
    "p2pPort": 12002,
    "pprofPort": 13002,
    "prefix": "akash",
    "rpcPort": 11002,
    "testnet-1": Object {
      "channelId": 12,
      "counterpartyChannelId": 66,
    },
  },
  "cosmos": Object {
    "binary": "gaiad",
    "binaryUrl": "https://github.com/cosmos/gaia/releases/download/v5.0.8/gaiad-v5.0.8-linux-amd64",
    "chainId": "cosmoshub-testnet",
    "denom": "uphoton",
    "devnet-1": Object {
      "channelId": 101,
      "counterpartyChannelId": 3,
    },
    "faucet": "https://faucet.testnet.cosmos.network",
    "fees": 0,
    "home": "/tmp/localnet/config/cosmos/cosmoshub-testnet",
    "node": "https://rpc.testnet.cosmos.network:443",
    "p2pPort": 12001,
    "pprofPort": 13001,
    "prefix": "cosmos",
    "rpcPort": 11001,
    "testnet-1": Object {
      "channelId": 11,
      "counterpartyChannelId": 27,
    },
  },
  "iris": Object {
    "binary": "iris",
    "chainId": "nyancat-8",
    "denom": "unyan",
    "devnet-1": Object {
      "channelId": 117,
      "counterpartyChannelId": 24,
    },
    "disabled": true,
    "fees": 400,
    "home": "/tmp/localnet/config/iris/nyancat-8",
    "node": "http://35.234.10.84:26657",
    "p2pPort": 12003,
    "pprofPort": 13003,
    "prefix": "iaa",
    "rpcPort": 11003,
    "sourceRelativePath": "irishub-1.1.1",
    "sourceUrl": "https://github.com/irisnet/irishub/archive/refs/tags/v1.1.1.zip",
    "testnet-1": Object {
      "channelId": 14,
      "counterpartyChannelId": 25,
      "receiver": Object {
        "denom": "ibc/7EABFE14CD2B6F8B0444A0ED0D7BCB947A81D0B74CE5B571C73DE61DFA261193",
        "fees": 150000,
        "gas": 300000,
      },
    },
  },
  "osmosis": Object {
    "binary": "osmosis",
    "binaryUrl": "https://github.com/osmosis-labs/osmosis/releases/download/v4.2.0/osmosisd-4.2.0-linux-amd64",
    "chainId": "osmosis-1",
    "denom": "uosmo",
    "devnet-1": Object {
      "channelId": 74,
      "counterpartyChannelId": 22,
    },
    "disabled": true,
    "fees": 0,
    "home": "/tmp/localnet/config/osmosis/osmosis-1",
    "node": "http://osmosis-rpc.osmosis-node:26657",
    "p2pPort": 12005,
    "pprofPort": 13005,
    "rpcPort": 11005,
    "testnet-1": Object {
      "channelId": 74,
      "counterpartyChannelId": 22,
    },
  },
  "persistence": Object {
    "binary": "persistenceCore",
    "binaryUrl": "https://github.com/persistenceOne/persistenceCore/releases/download/v0.1.3/persistenceCore-linux-amd64.tar.gz",
    "chainId": "test-core-1",
    "denom": "uxprt",
    "devnet-1": Object {
      "channelId": 120,
      "counterpartyChannelId": 25,
    },
    "fees": 100000000000,
    "home": "/tmp/localnet/config/persistence/test-core-1",
    "node": "https://persistence.testnet.rpc.audit.one:443",
    "p2pPort": 12006,
    "pprofPort": 13006,
    "prefix": "persistence",
    "rpcPort": 11006,
    "testnet-1": Object {
      "channelId": 15,
      "counterpartyChannelId": 24,
      "receiver": Object {
        "denom": "ibc/2915CCA550DA6840CB2D4F5BC95A067A7924A767B2EBEB3227DD66124AA8A626",
        "fees": 150000,
        "gas": 300000,
      },
    },
  },
  "sentinel": Object {
    "binary": "sentinelhub",
    "chainId": "sentinelhub-2",
    "denom": "udvpn",
    "devnet-1": Object {
      "channelId": 79,
      "counterpartyChannelId": 19,
    },
    "disabled": true,
    "fees": 20000,
    "home": "/tmp/localnet/config/sentinel/sentinelhub-2",
    "node": "http://rpc.sentinel.co:26657",
    "p2pPort": 12004,
    "pprofPort": 13004,
    "prefix": "sent",
    "rpcPort": 11004,
    "sourceRelativePath": "sentinel-0.1.4",
    "sourceUrl": "https://github.com/sentinel-official/sentinel/archive/refs/tags/v0.1.4.zip",
    "testnet-1": Object {
      "channelId": 39,
      "counterpartyChannelId": 2,
    },
  },
  "sifchain": Object {
    "betanet": Object {
      "chainId": "sifchain-1",
      "node": "https://rpc.sifchain.finance:443",
    },
    "binary": "sifnoded",
    "binaryUrl": "https://github.com/Sifchain/sifnode/releases/download/betanet-0.9.10/sifnoded-betanet-0.9.10-linux-amd64.zip",
    "chainId": "sifchain-testnet-1",
    "denom": "rowan",
    "devnet-1": Object {
      "chainId": "sifchain-devnet-1",
      "node": "https://rpc-devnet.sifchain.finance:443",
    },
    "fees": 100000,
    "home": "/tmp/localnet/config/sifchain/sifchain-testnet-1",
    "node": "https://rpc-testnet.sifchain.finance:443",
    "p2pPort": 12000,
    "pprofPort": 13000,
    "prefix": "sif",
    "rpcPort": 11000,
    "testnet-1": Object {
      "chainId": "sifchain-testnet-1",
      "node": "https://rpc-testnet.sifchain.finance:443",
    },
  },
  "terra": Object {
    "binary": "terrad",
    "binaryUrl": "https://github.com/terra-money/core/releases/download/v0.5.10/terra_0.5.10_Linux_x86_64.tar.gz",
    "chainId": "bombay-10",
    "denom": "uluna",
    "devnet-1": Object {
      "channelId": 123,
      "counterpartyChannelId": 3,
    },
    "disabled": true,
    "fees": 0,
    "home": "/tmp/localnet/config/terra/bombay-10",
    "node": "http://3.34.120.243:26657",
    "p2pPort": 12007,
    "pprofPort": 13007,
    "prefix": "terra",
    "rpcPort": 11007,
    "testnet-1": Object {
      "channelId": 29,
      "counterpartyChannelId": 4,
    },
  },
}
`;
